projects/plugins/jetpack/modules/masterbar/admin-menu/dashicon-set.php

```diff
@@ -0,0 +1,358 @@
+	'dashicons-star-filled'               => true,
+	'dashicons-star-half'                 => true,
+	'dashicons-sticky'                    => true,
+	'dashicons-store'                     => true,
+	'dashicons-superhero-alt'             => true,
+	'dashicons-superhero'                 => true,
+	'dashicons-table-col-after'           => true,
+	'dashicons-table-col-before'          => true,
+	'dashicons-table-col-delete'          => true,
+	'dashicons-table-row-after'           => true,
+	'dashicons-table-row-before'          => true,
+	'dashicons-table-row-delete'          => true,
+	'dashicons-tablet'                    => true,
+	'dashicons-tag'                       => true,
+	'dashicons-tagcloud'                  => true,
+	'dashicons-testimonial'               => true,
+	'dashicons-text-page'                 => true,
+	'dashicons-text'                      => true,
+	'dashicons-thumbs-down'               => true,
+	'dashicons-thumbs-up'                 => true,
+	'dashicons-tickets-alt'               => true,
+	'dashicons-tickets'                   => true,
+	'dashicons-tide'                      => true,
+	'dashicons-translation'               => true,
+	'dashicons-trash'                     => true,
+	'dashicons-twitch'                    => true,
+	'dashicons-twitter-alt'               => true,
+	'dashicons-twitter'                   => true,
+	'dashicons-undo'                      => true,
+	'dashicons-universal-access-alt'      => true,
+	'dashicons-universal-access'          => true,
+	'dashicons-unlock'                    => true,
+	'dashicons-update-alt'                => true,
+	'dashicons-update'                    => true,
+	'dashicons-upload'                    => true,
+	'dashicons-vault'                     => true,
+	'dashicons-video-alt'                 => true,
+	'dashicons-video-alt2'                => true,
+	'dashicons-video-alt3'                => true,
+	'dashicons-visibility'                => true,
+	'dashicons-warning'                   => true,
+	'dashicons-welcome-add-page'          => true,
+	'dashicons-welcome-comments'          => true,
+	'dashicons-welcome-learn-more'        => true,
+	'dashicons-welcome-view-site'         => true,
+	'dashicons-welcome-widgets-menus'     => true,
+	'dashicons-welcome-write-blog'        => true,
+	'dashicons-whatsapp'                  => true,
+	'dashicons-wordpress-alt'             => true,
+	'dashicons-wordpress'                 => true,
+	'dashicons-xing'                      => true,
+	'dashicons-yes-alt'                   => true,
+	'dashicons-yes'                       => true,
+	'dashicons-youtube'                   => true,
+	'dashicons-editor-distractionfree'    => true,
+	'dashicons-exerpt-view'               => true,
+	'dashicons-format-links'              => true,
+	'dashicons-format-standard'           => true,
+	'dashicons-post-trash'                => true,
+	'dashicons-share1'                    => true,
+	'dashicons-welcome-edit-page'         => true,
+);
```

Comment:
Chiming in on this a bit late, but still! 🙂 

Is there a particular reason for doing this "fake set" structure (as you [mentioned elsewhere](https://github.com/Automattic/jetpack/pull/22070#discussion_r767524308)), instead of a simple keyless array and looking for icons with `in_array`?

It's not a big deal, but I agree with others that the boolean value seems to hint that icons might be turned on and off — which: it is possible with the current `isset( $this->dashicon_list[ $icon ] ) && $this->dashicon_list[ $icon ]` check, but not really something that we would do. 🤔

Reply:
@Copons  It's just for performance reasons, in_array has a time complexity of O(n) whereas searching in a set has time complexity O(1).

Reply:
> It's just for performance reasons

@rcrdortiz Sounds good to me then! ✨ 

> > but I agree with others that the boolean value seems to hint that icons might be turned on and off — which: it is possible with the current isset( $this->dashicon_list[ $icon ] ) && $this->dashicon_list[ $icon ]
> 
> I also don't like that check, I've added it by popular demand.

😄 
Just to clarify, I don't have a strong opinion on the check itself.
I simply wanted to point out how that check, written that way, might trick people into misunderstanding the icon set's purpose.
This said, since we are going with the set, then IMHO it makes sense to go all-in and support conditionally disabling icons (e.g. `'dashicons-foobar' => false`) in the set.
We don't have a use-case for that yet, but it makes the code more consistent and predictable.